src/camera.c

```diff
@@ -126,7 +126,7 @@ query_permission_sync (Request *request)
 
       if (!xdp_dbus_impl_access_call_access_dialog_sync (access_impl,
                                                          request->id,
-                                                         app_id,
+                                                         xdp_is_valid_app_id (app_id) ? app_id : "",
```

Comment:
Isn't it the point? `org.kde.kcalc` is valid, because the appstream id requires to be in form of `domain.organization.app_name` so we want to avoid using simple names like `vlc` or `firefox`.

Reply:
That is only enforced for flaptak (and snap, I guess), but not the rest, are they not. Not sure making such enforcements here is the best place.

Reply:
I understand, however, using an invalid app id is as useless as sending an empty one, except that the backend assumes it's valid and rejects the request based on this assumption.  

Do you want me to change it back to the original code? I tested the scenario with Firefox from Flathub and it found the desktop file after installation and identified the app correctly.

Reply:
> Do you want me to change it back to the original code? I tested the scenario with Firefox from Flathub and it found the desktop file after installation and identified the app correctly.

Yea, after another round of thought, the original idea you had seems indeed better.

Reply:
Though, something I did *not* check was whether uninstalling a Flatpak removes the desktop file with the app still running, and for how long the app info is in the cache.